### ISSUE TITLE
PeerStatus.LastHandshake isn't what you might think

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#1182]
I tried to use `PeerStatus.LastHandshake`:

```
type PeerStatus struct {
        TxBytes, RxBytes ByteCount
        LastHandshake    time.Time                                                                                                                                                                                                                               
        NodeKey          tailcfg.NodeKey
}
```

Given the lack of docs, I had to assume it meant "the time the handshake last completed".

Nope. It apparently means "the last time we tried to start a handshake, even if the peer was offline". Which isn't very useful.

Document either way.

Ideally also add the thing we actually want for `wgengine/pendopen.go`.

/cc @josharian @crawshaw

**Comment by josharian** [comment]
> Nope. It apparently means "the last time we tried to start a handshake, even if the peer was offline". Which isn't very useful.

I looked through wireguard, and it appears to mean "the last time the handshake completed or we got key confirmation via the first data message". (We get it from the last_handshake_time_sec/last_handshake_time_nsec uapi fields, which are populated from peer.stats.lastHandshakeNano, which is updated in timersHandshakeComplete, which is called from RoutineSequentialReceiver when receiving key confirmation and from RoutineHandshake when we receive a validated handshake response.)

I'll send a docs PR shortly.

But outstanding questions are: Why did you come to the conclusion that this field didn't mean what it appeared to (and did) mean? And should I leave this bug open for changing pendopen.go?
